### PR TITLE
Refactor primitive geometry

### PR DESCRIPTION
**Pull request by singlecoder**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
refactor PrimitiveMesh
### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
add new api named createCylinder for create cylinder

### CODE REVIEW

File: packages/core/src/mesh/PrimitiveMesh.ts [hunk 1]
```diff
@@ -27,69 +26,73 @@ export class PrimitiveMesh {
...
     engine: Engine,
     radius: number = 1,
-    horizontalSegments: number = 8,
-    verticalSegments: number = 6,
+    horizontalSegments: number = 12,
+    verticalSegments: number = 12,
     alphaStart: number = 0,
     alphaRange: number = Math.PI * 2,
     thetaStart: number = 0,
     thetaRange: number = Math.PI
```
Comment (GuoLei1990) [inline packages/core/src/mesh/PrimitiveMesh.ts:34]: alphaStart、alphaRange、thetaStart、thetaRange maybe don't have practical value.

**Review by GuoLei1990 — CHANGES_REQUESTED** [review]

File: packages/core/src/mesh/PrimitiveMesh.ts (hunk 1 above)
Comment (singlecoder) [reply packages/core/src/mesh/PrimitiveMesh.ts]: > alphaStart、alphaRange、thetaStart、thetaRange maybe don't have practical value.

has deleted, keep simple

File: packages/core/src/mesh/PrimitiveMesh.ts [hunk 2]
```diff
@@ -26,70 +25,69 @@ export class PrimitiveMesh {
...
+      const u = x / segments;
+      const v = y / segments;
+      const alphaDelta = u * alphaRange;
+      const thetaDelta = v * thetaRange;
+      const sinTheta = Math.sin(thetaDelta);
+
+      let posX = -radius * Math.cos(alphaDelta) * sinTheta;
+      let posY = radius * Math.cos(thetaDelta);
+      let posZ = radius * Math.sin(alphaDelta) * sinTheta;
+      posX = Math.abs(posX) < 1e-6 ? 0 : posX;
```
Comment (GuoLei1990) [inline packages/core/src/mesh/PrimitiveMesh.ts]: use MathUtil.zeroTolerance to instead of 1e-6

File: packages/core/src/mesh/PrimitiveMesh.ts [hunk 3]
```diff
@@ -158,112 +178,187 @@ export class PrimitiveMesh {
...
+      const u = x / radialSegments;
+      const v = y / heightSegments;
+      const theta = u * Math.PI * 2;
+      const sinTheta = Math.sin(theta);
+      const cosTheta = Math.cos(theta);
+      
+      let posX = radius * sinTheta;
+      let posY = v * unitHeight - halfHeight;
+      let posZ = radius * cosTheta;
+      posX = Math.abs(posX) < 1e-6 ? 0 : posX;
```
Comment (GuoLei1990) [inline packages/core/src/mesh/PrimitiveMesh.ts]: Math.abs(posX) < 1e-6 can delete

File: packages/core/src/mesh/PrimitiveMesh.ts [hunk 4]
```diff
@@ -158,112 +178,187 @@ export class PrimitiveMesh {
...
+    let verticesOffset = 0;
+    let indicesOffset = 0;
+    
+    // Create torso
+    for (let i = 0; i < torsoVerticesCount; ++i) {
+      const x = i % radialCount;
+      const y = i / radialCount | 0;
+      const u = x / radialSegments;
+      const v = y / heightSegments;
+      const theta = u * Math.PI * 2;
```
Comment (GuoLei1990) [inline packages/core/src/mesh/PrimitiveMesh.ts:268]: Math.PI * 2 should cache brfore for(;;)

**Review by GuoLei1990 — COMMENTED** [review]
createSphere

**Review by GuoLei1990 — APPROVED** [review]
+1

**Pull request merged**